contracts/HATKlerosConnector.sol

```diff
@@ -0,0 +1,486 @@
+        ExecuteResolution, // Accept the claim as it is without changing.
+        DismissResolution // Dismiss the claim.
+    }
+
+    struct DisputeStruct {
+        bytes32 claimId; // Id of the claim in HATVault contract.
+        uint256 externalDisputeId; // Id of the dispute created in Kleros court.
+        Decision ruling; // Ruling given by the arbitrator.
+        bool resolved; // True if the dispute has been resolved.
+        Round[] rounds; // Appeal rounds.
+    }
+
+    // Round struct stores the contributions made to particular sides.
+    // - 0 side for `Decision.None`.
+    // - 1 side for `Decision.ExecuteResolution`.
+    // - 2 side for `Decision.DismissResolution`.
+    struct Round {
+        uint256[3] paidFees; // Tracks the fees paid in this round in the form paidFees[side].
+        bool[3] hasPaid; // True if the fees for this particular side have been fully paid in the form hasPaid[side].
+        mapping(address => uint256[3]) contributions; // Maps contributors to their contributions for each side in the form contributions[address][side].
+        uint256 feeRewards; // Sum of reimbursable appeal fees available to the parties that made contributions to the side that ultimately wins a dispute.
+        uint256[] fundedSides; // Stores the sides that are fully funded.
+    }
+
+    address public immutable governor; // Governor of this contract.
+    IArbitrator public immutable klerosArbitrator; // The kleros arbitrator contract (e.g. Kleros Court).
+    HATArbitrator public immutable hatArbitrator; // Address of the Hat arbitrator contract.
+    IHATVault public immutable vault; // Vault contract.
+    uint256 public metaEvidenceUpdates; // Relevant index of the metaevidence.
+    bytes public arbitratorExtraData; // Extra data for the arbitrator.
+
+    uint256 public winnerMultiplier; // Multiplier for calculating the appeal fee that must be paid for the ruling that was chosen by the arbitrator in the previous round, in basis points.
+    uint256 public loserMultiplier; // Multiplier for calculating the appeal fee that must be paid for the ruling that the arbitrator didn't rule for in the previous round, in basis points.
+    uint256 public loserAppealPeriodMultiplier; // Multiplier for calculating the duration of the appeal period for the loser, in basis points.
+
+    DisputeStruct[] public disputes; // Stores the disputes created in this contract.
+    mapping(bytes32 => bool) public claimChallenged; // True if the claim was challenged in this contract..
+    mapping(uint256 => uint256) public override externalIDtoLocalID; // Maps external dispute ids to local dispute ids.
+
+    /** @dev Raised when a claim is challenged.
+     *  @param _claimId Id of the claim in Vault cotract.
+     */
+    event Challenged(bytes32 indexed _claimId);
+    
+    modifier onlyGovernor {require(msg.sender == governor, "The caller must be the governor."); _;}
```

Comment:
I think it will be better to use Ownable here instead of this. We might want to allow transferring ownership.